File: Data Structures & Algorithms/string-encode-and-decode/submission-0.cpp
class Solution {
public:

    string encode(vector<string>& strs) {
        string encoded = "";
        for(string word: strs){
            encoded += to_string(word.length()) + "#" +word;
        }
        cout<<encoded;
        return encoded;
    }

    vector<string> decode(string s) {
        vector<string> res;
        int itr = 0;
        while(itr < s.length()){
            string num = "";
            while(s[itr] != '#'){
                num += s[itr];
                itr++;
            }

            itr++;
            res.push_back(s.substr(itr, stoi(num)));
            itr += stoi(num);
        }
        return res;
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        map={}
        l=0
        res = 0
        for r,ch in enumerate(s):
            if ch in map:
                if l<=map[ch]:
                    l=map[ch]+1
            map[ch] = r
            res = max(res, r-l+1)
        return res

            
        